scripts/maintenance/create_client_credentials.pl

```diff
@@ -69,8 +77,8 @@ sub main {
 		say "Client secret: $client_secret";
 	}
 	if ( $opts{'i'} || $opts{'u'} ) {
-		my $db = DBI->connect( 'DBI:Pg:dbname=' . DBASE,
-			'postgres', '', { AutoCommit => 0, RaiseError => 1, PrintError => 0 } )
+		my $db = DBI->connect( 'DBI:Pg:dbname=' . DBASE . ";host=$opts{'dbhost'};port=$opts{'dbport'}",
```

Comment:
A problem here is that previously you could run this command as the postgres user on localhost without having to provide a password. Now, if you don't use the -P option, it fails with:

DBI connect('dbname=bigsdb_auth;host=localhost;port=5432','postgres',...) failed: fe_sendauth: no password supplied at ./create_cred.pl line 80.

I 'sudo su postgres' and then run the command. I think we need a test so that if no password is provided that you don't attempt to pass an empty string to the connect method.